### ISSUE TITLE
[bug] Setting default window to as invisible in the config it leads to Tauri crashing on Linux.

### ISSUE THREAD

**Issue by fayeed** [tauri-apps/tauri#4149]
### Describe the bug

I am trying to have the window hidden by default while I spawn a new process and then show the window but on Linux, if you set the default window to be invisible in the config it leads to Tauri crashing, works fine on Windows & macOS.

### Reproduction

1. Set visible property in windows to false & run the app
```json
 "windows": [
      {
        "label": "main",
        "title": "main",
        "width": 1000,
        "height": 700,
        "resizable": true,
        "fullscreen": false,
        "center": true,
        "focus": true,
       "visible": false
      }
],
```

### Expected behavior

To have the default window stay invisible until it is set to visible without crashing.

### Platform and versions

```shell
Environment
  › OS: Ubuntu 22.04 X64
  › Node.js: 14.16.0
  › npm: 8.1.2
  › pnpm: Not installed!
  › yarn: 1.22.10
  › rustup: 1.24.3
  › rustc: 1.60.0
  › cargo: 1.60.0
  › Rust toolchain: stable-x86_64-unknown-linux-gnu 

Packages
  › @tauri-apps/cli [NPM]: 1.0.0-rc.10
  › @tauri-apps/api [NPM]: 1.0.0-rc.3
  › tauri [RUST]: 1.0.0-rc.10,
  › tauri-build [RUST]: 1.0.0-rc.8,
  › tao [RUST]: 0.8.4,
  › wry [RUST]: 0.16.2,

App
  › build-type: bundle
  › CSP: unset
  › distDir: ../build
  › devPath: http://localhost:3000/
  › framework: React

App directory structure
  ├─ node_modules
  ├─ src
  ├─ scripts
  ├─ .git
  ├─ .github
  ├─ src-tauri
  ├─ build
  └─ public
```

### Stack trace

```shell
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', /home/fayeed/.cargo/registry/src/github.com-1ecc6299db9ec823/tao-0.8.4/src/platform_impl/linux/window.rs:602:39
stack backtrace:
   0: rust_begin_unwind
             at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/panicking.rs:584:5
   1: core::panicking::panic_fmt
             at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/core/src/panicking.rs:143:14
   2: core::panicking::panic
             at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/core/src/panicking.rs:48:5
   3: core::option::Option<T>::unwrap
             at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/core/src/option.rs:752:21
   4: tao::platform_impl::platform::window::Window::current_monitor
             at /home/fayeed/.cargo/registry/src/github.com-1ecc6299db9ec823/tao-0.8.4/src/platform_impl/linux/window.rs:602:18
   5: tao::window::Window::current_monitor
             at /home/fayeed/.cargo/registry/src/github.com-1ecc6299db9ec823/tao-0.8.4/src/window.rs:914:5
   6: tauri_runtime_wry::center_window
             at /home/fayeed/.cargo/registry/src/github.com-1ecc6299db9ec823/tauri-runtime-wry-0.5.1/src/lib.rs:2857:26
   7: tauri_runtime_wry::create_webview
             at /home/fayeed/.cargo/registry/src/github.com-1ecc6299db9ec823/tauri-runtime-wry-0.5.1/src/lib.rs:2941:13
   8: <tauri_runtime_wry::Wry<T> as tauri_runtime::Runtime<T>>::create_window
             at /home/fayeed/.cargo/registry/src/github.com-1ecc6299db9ec823/tauri-runtime-wry-0.5.1/src/lib.rs:1936:19
   9: tauri::app::Builder<R>::build
             at /home/fayeed/.cargo/registry/src/github.com-1ecc6299db9ec823/tauri-1.0.0-rc.10/src/app.rs:1418:22
  10: ad4min::main
             at ./src/main.rs:59:26
  11: core::ops::function::FnOnce::call_once
             at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/core/src/ops/function.rs:227
```

**Comment by FabianLars** [comment]
Does somebody got a better idea than this?
1) Don't panic in tao's `current_monitor` function, return None instead (Unless this is a bug and it's supposed to have a gtk window).
2) Maybe use the primary monitor as a fallback if current_monitor returns None? I don't know if that would be 

And, unrelated but annoying, `center()` doesn't account the monitor position which leads to the window being moved to the leftmost monitor (and with different monitor sizes it ends up not being centered). I think the fix could be as easy as this?
```rs
// Inside center_window() in tauri-runtime-wry
#[cfg(target_os = "linux")]
window.set_outer_position(WryPhysicalPosition::new(
        monitor_pos.x + x,
        monitor_pos.y + y,
));
```

**Comment by lucasfernog** [comment]
Fixed in tao 0.8.5, to use that just run `cd src-tauri && cargo update -p tao` (still publishing, will be ready in a few minutes). I'll fix the `center` API in a separate PR.
